bc_obps/service/report_service.py

```diff
@@ -85,22 +90,26 @@ def get_report_operation_by_version_id(cls, report_version_id: int) -> ReportOpe
     def save_report_operation(cls, report_version_id: int, data: ReportOperationIn) -> ReportOperation:
         # Fetch the existing report operation
         report_operation = ReportOperation.objects.get(report_version__id=report_version_id)
+        report_operation_representatives = ReportOperationRepresentative.objects.filter(
+            report_version=report_version_id
+        )
 
+        # Update the selected_for_report field based on the provided data
+        representative_ids_in_data = data.operation_representative_name  # List of IDs from input
+        for representative in report_operation_representatives:
+            representative.selected_for_report = representative.id in representative_ids_in_data
+            representative.save()
```

Comment:
We can write this in a more django way:
```
ReportOperationRepresentative.objects.filter(report_version_id=report_version_id, id__in=representative_ids_in_data).update(selected_for_report=True)
ReportOperationRepresentative.objects.filter(report_version_id=report_version_id).exclude(id__in=representative_ids_in_data).update(selected_for_report=False)
```

Reply:
@Sepehr-Sobhani might have a way to bake that into one single `.update(...)` with a subquery